### PR TITLE
[FIX] l10n_ar_edi_ux: We consider whether the surname or first name data is empty

### PR DESCRIPTION
**Pull request by mem-adhoc**
Task: 29136

### CODE REVIEW

**Comment by roboadhoc** [comment]
[Pull request status dashboard](https://mergebot.adhoc.ar/ingadhoc/odoo-argentina-ee/pull/363).

File: l10n_ar_edi_ux/models/res_partner.py [hunk 1]
```diff
@@ -71,29 +77,41 @@ def get_data_from_padron_afip(self):
...
             raise UserError(error_msg % (self.name, vat, 'La afip no devolvió nombre'))
 
+        self._clean_dict(data.get("domicilioFiscal"), type_replace={})
         domicilio = data.get("domicilioFiscal", {})
         data_mt = zeep.helpers.serialize_object(res.datosMonotributo, dict) or {}
         data_rg = zeep.helpers.serialize_object(res.datosRegimenGeneral, dict) or {}
+
+        self._clean_dict(data_mt.get("impuesto"), type_replace=[])
+        import pdb
+        pdb.set_trace()
```
Comment (zaoral) [inline l10n_ar_edi_ux/models/res_partner.py]: remove

File: l10n_ar_edi_ux/models/res_partner.py [hunk 2]
```diff
@@ -35,6 +35,12 @@ def update_constancia_from_padron_afip(self):
         self.ensure_one()
         return True
 
+    def _clean_dict(self, dictionary, type_replace=''):
+        #Limpiamos los diccionarios para quitarle los valores None
+        if dictionary and type(dictionary) == dict:
+            for val in dictionary:
+                dictionary[val] = type_replace if not dictionary[val] else dictionary[val]
```
Comment (zaoral) [inline l10n_ar_edi_ux/models/res_partner.py]: only check the element, and replace it with the type_replace that match if None

File: l10n_ar_edi_ux/models/res_partner.py [hunk 3]
```diff
@@ -128,6 +146,7 @@ def check_taxes(data_mt, data_rg):
             return res
         check_taxes(data_mt, data_rg)
 
+        self._clean_dict(data_rg.get("categoriaMonotributo"), type_replace={})
         cat_mt = data_mt.get("categoriaMonotributo", {})
```
Comment (zaoral) [inline l10n_ar_edi_ux/models/res_partner.py]: need to remove second line,         cat_mt = elf._clean_dict(data_rg.get("categoriaMonotributo"), type_replace={})

File: l10n_ar_edi_ux/models/res_partner.py [hunk 4]
```diff
@@ -71,29 +77,41 @@ def get_data_from_padron_afip(self):
...
+        import pdb
+        pdb.set_trace()
+        self._clean_dict(data_rg.get("impuesto"), type_replace=[])
+
         impuestos = [imp["idImpuesto"]
                      for imp in data_mt.get("impuesto", []) + data_rg.get("impuesto", [])
                      if data.get('estadoClave') == 'ACTIVO']
 
+        self._clean_dict(data_mt.get("actividadMonotributista"), type_replace=[])
         data_mt_actividades = data_mt.get("actividadMonotributista", []) or []
```
Comment (zaoral) [inline l10n_ar_edi_ux/models/res_partner.py]: fusionar

File: l10n_ar_edi_ux/models/res_partner.py [hunk 5]
```diff
@@ -71,29 +77,41 @@ def get_data_from_padron_afip(self):
...
 
+        self._clean_dict(data.get("domicilioFiscal"), type_replace={})
         domicilio = data.get("domicilioFiscal", {})
         data_mt = zeep.helpers.serialize_object(res.datosMonotributo, dict) or {}
         data_rg = zeep.helpers.serialize_object(res.datosRegimenGeneral, dict) or {}
+
+        self._clean_dict(data_mt.get("impuesto"), type_replace=[])
+        import pdb
+        pdb.set_trace()
+        self._clean_dict(data_rg.get("impuesto"), type_replace=[])
```
Comment (zaoral) [inline l10n_ar_edi_ux/models/res_partner.py]: esta repetido

File: l10n_ar_edi_ux/models/res_partner.py [hunk 6]
```diff
@@ -71,29 +77,41 @@ def get_data_from_padron_afip(self):
...
             raise UserError(error_msg % (self.name, vat, res))
+        self._clean_dict(data, type_replace=[])
 
-        denominacion = data.get("razonSocial", "") or ", ".join([data.get("apellido", ""), data.get("nombre", "")])
+        denominacion = data.get("razonSocial", "") or ", ".join([str(data.get("apellido", "")), str(data.get("nombre", ""))])
         if not denominacion or denominacion == ', ':
             raise UserError(error_msg % (self.name, vat, 'La afip no devolvió nombre'))
 
+        self._clean_dict(data.get("domicilioFiscal"), type_replace={})
         domicilio = data.get("domicilioFiscal", {})
```
Comment (zaoral) [inline l10n_ar_edi_ux/models/res_partner.py]: fusionar

**Review by zaoral — CHANGES_REQUESTED** [review]

File: l10n_ar_edi_ux/models/res_partner.py (hunk 2 above)
Comment (mem-adhoc) [reply l10n_ar_edi_ux/models/res_partner.py]: Here we take the dictionary and really check if it has any None element, if it is None we replace it.

File: l10n_ar_edi_ux/models/res_partner.py (hunk 6 above)
Comment (mem-adhoc) [reply l10n_ar_edi_ux/models/res_partner.py]: _clean_dict does not return anything, check if the dictionary had a None value and replace it. What do you mean by fusion?

**Comment by zaoral** [comment]
@roboadhoc r+

**Comment by fw-bot-adhoc** [comment]
@mem-adhoc @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina-ee#501

**Comment by fw-bot-adhoc** [comment]
@mem-adhoc @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina-ee#501

**Comment by fw-bot-adhoc** [comment]
@mem-adhoc @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina-ee#501

**Comment by fw-bot-adhoc** [comment]
@mem-adhoc @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina-ee#501

**Comment by fw-bot-adhoc** [comment]
@mem-adhoc @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina-ee#501

**Comment by fw-bot-adhoc** [comment]
@mem-adhoc @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina-ee#501

**Comment by fw-bot-adhoc** [comment]
@mem-adhoc @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina-ee#501

**Comment by fw-bot-adhoc** [comment]
@mem-adhoc @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina-ee#501

**Pull request closed without merging**